python/vowpalwabbit/DFtoVWtoTensorboard.py

```diff
@@ -0,0 +1,208 @@
+
+		except ZeroDivisionError:
+			return 0.
+
+
+	#------------------------------------------------------------------------------------------------		
+	def _print_metrics(self, average_loss, since_last, label, prediction, num_features):
+		"""This method prints metrics
+
+		Parameters
+		----------
+
+		average_loss : float
+						This is the average_loss value
+		since_last   : float
+						This is the since_last value
+		label        : integer
+						This is the actual label of current example
+		prediction   : integer
+						This is the predicted label of current example
+		num_features : integer
+						This is the num_features value
+
+		Returns
+		-------
+
+		None
+		"""
+		print( 'average_loss:{:.6f}'.format(average_loss) , end='\t')
+		print('since_last:{:.6f}'.format(since_last), end='\t')
+		print('label:', label, end='\t')    
+		print('prediction:', prediction, end='\t')    
+		print('num_features:', num_features) 
+
+
+	#------------------------------------------------------------------------------------------------		
+	def get_label(self, example, label_type):
+		"""Similar to pyvw.get_prediction implementation this method gets the label for current example
+
+		Parameters
+		----------
+
+		example    : example object
+						Methods of this example object are used to get the appropiate label
+		label_type : integer
+						This is the integer for representing a specific type of label in vw
+						- 0: lDEFAULT
+						- 1: lBINARY
+						- 2: lMULTICLASS
+						- 3: lCOST_SENSITIVE
+						- 4: lCONTEXTUAL_BANDIT
+						- 5: lMAX
+						- 6: lCONDITIONAL_CONTEXTUAL_BANDIT
+						- 7: lSLATES
+						- 8: lCONTINUOUS
+
+		Returns
+		-------
+
+		label  : integer
+					Actual label of the current example object
+		"""
+		switch_label_type = {
+			pyvw.vw.lDefault: None,
+			pyvw.vw.lBinary: example.get_simplelabel_label,
+			pyvw.vw.lMulticlass: example.get_multiclass_label,
+			pyvw.vw.lCostSensitive: example.get_costsensitive_class,
+			pyvw.vw.lContextualBandit: example.get_cbandits_class,
+	#         pyvw.vw.lConditionalContextualBandit: None,
+	#         pyvw.vw.lSlates: None,
+	#         pyvw.vw.lContinuous: None
+		}
+		return switch_label_type[label_type]()	
+
+
+	#------------------------------------------------------------------------------------------------		
+	def fit(self, df, tensorboard=True):
+		"""Learns on the relevant examples and can also log metrics for tensorboard visualization
+
+		Parameters
+		----------
+
+		df           : Pandas Dataframe object
+						This is used for vw to learn from the corresponding examples of the records in this Pandas Dataframe object
+		tensorboard  : boolean
+						Default value is True, this parameter is used to control the logging of metrics for Tensorboard visualization
+						If value is True : metrics are logged for Tensorboard visualization
+						If value is False : metrics are not logged for Tensorboard visualization
+
+		Returns
+		-------
+
+		None
+		"""
+		if tensorboard:
```

Comment:
Consider wrapping this bit into a VWtoTensorboard class, and optionally pass it to your .learn/.fit methods, rather than a flag to turn Tensorboard on. This will allow you to specify the log folder rather than hardcoding it, and later configure which metrics are projected into Tensorboard.

Reply:
I realized that the above is a bit ambiguous: If switching to a VWtoTensorboard class, you should enable/disable Tensorboard export based on whether an instance of the class was passed in to your `.learn`/`.fit` method. On this class you can set up things like what the appropriate log paths are, run identifiers, etc.

Reply:
Okay I would try that now